lmfit/minimizer.py

```diff
@@ -282,11 +290,16 @@ def __residual(self, fvars):
         if self._abort:
             return None
         params = self.result.params
-        for name, val in zip(self.result.var_names, fvars):
-            params[name].value = params[name].from_internal(val)
+
+        if apply_scaling:
+            for name, val in zip(self.result.var_names, fvars):
+                params[name].value = params[name].from_internal(val)
+                params.update_constraints()
+        else:
+            for name, val in zip(self.result.var_names, fvars):
+                params[name].value = val
```

Comment:
You still need to call `params.update_constraints`, even if `apply_scaling is False`. This method is used to make sure that all `Parameter.expr` are evaluated. It's worth adding a test to see if `least_squares` works with expressions.